README.md

```diff
@@ -1,3 +1,78 @@
-# Tracking ML
+# Tracking MLflow jobs on Openshift
 
-Demo of tracking machine learning metrics into mlflow running on k8s
+This is a demo application which includes templates that can be used to run mlflow jobs on Openshift.
+
+## Running on OpenShift
+This is a two stage process, because you need to create the image only once and then you can create multiple containers using the same image but with different parameters.
+
+### Create a container image
+To run this application on openshift, we need to create a container image for it.  To do this we have a image build template [here in the repository](https://github.com/AICoE/experiment-tracking-template/blob/master/openshift/mlflow-experiment-image-buildconfig.yaml).
+
+This template will take a base image, and install all the dependencies listed in the Pipfile and store this new image to the image registry.
+
+To initiate a new image build, run the following command from the root of this repository:
+
+```
+oc new-app ./openshift/mlflow-experiment-image-buildconfig.yaml --param APPLICATION_NAME=mlflow-experiment-image-demo --param GIT_URI=https://github.com/AICoE/experiment-tracking-template.git --env APP_FILE=app.py
```

Comment:
so the flow would be 

1. admin creates template in openshift via `oc create template.yaml`
2. DS uses template to create his BC via `oc process mlflow-experiment-bc | oc create -f -`

I think `oc process` is better here, because `new-app` does way more than we need

Reply:
@4n4nd You can use this sample mlflow project to demonstrate building the image. Also note that there are a few environment variables you want to set:

* `MLFLOW_TRACKING_URI`
* `MLFLOW_EXPERIMENT_ID`

Reply:
Try to use this https://github.com/zmhassan/sample-mlflow-tracking-demo git repo to demo it running.

Reply:
this repo here is a fork of yours :)
and the MLFLOW_TRACKING_URI and MLFLOW_EXPERIMENT_ID should be defaulted. So that the bar to run something is reaaaally low

Reply:
Instead of MLFLOW_EXPERIMENT_ID, I am planning to use just MLFLOW_EXPERIMENT_NAME which should be easier to use for DSs